### PR TITLE
Controller fixes

### PR DESCRIPTION
**Pull request by csrwng**
- Always clear channel and upstream on ClusterVersion resource (upgrades will be driven from the control plane)
- Change oauth restarter controller to watch the kubeadmin secret in the child cluster and use its content to trigger a redeploy of the oauth server on the control plane.

### CODE REVIEW

**Comment by csrwng** [comment]
@joelddiaz ptal

File: pkg/controllers/kubeadminpwd/oauthrestarter.go [hunk 1]
```diff
@@ -1,77 +1,98 @@
...
+	if req.Name != KubeAdminSecret {
 		return ctrl.Result{}, nil
 	}
 
 	controllerLog.Info("Begin reconciling")
 
-	pod := &corev1.Pod{}
-	if err := o.Get(ctx, req.NamespacedName, pod); err != nil {
+	secret, err := o.SecretLister.Secrets(metav1.NamespaceSystem).Get(KubeAdminSecret)
+	if err != nil && errors.IsNotFound(err) {
```
Comment (csrwng) [inline pkg/controllers/kubeadminpwd/oauthrestarter.go:53]: This is wrong

**Comment by csrwng** [comment]
Thx @joelddiaz ! I addressed your comments and added unit tests

File: pkg/controllers/kubeadminpwd/oauthrestarter.go [hunk 2]
```diff
@@ -1,77 +1,93 @@
...
-	// Ignore any namespace that is not the Namespace.
-	if req.Namespace != o.Namespace {
-		return ctrl.Result{}, nil
-	}
+	controllerLog := o.Log.WithValues("secret", req.NamespacedName.String())
 
-	// Ignore all pods except the manifest bootstrapper
-	if req.Name != ManifestBootstrapperPod {
+	// Ignore any secret that is not kube-system/kubeadmin
+	if req.Namespace != metav1.NamespaceSystem || req.Name != KubeAdminSecret {
```
Comment (joelddiaz) [inline pkg/controllers/kubeadminpwd/oauthrestarter.go:41]: should be `&&` as we don't want secret kube-system/ignore-this-secret to get through

Comment (joelddiaz) [reply pkg/controllers/kubeadminpwd/oauthrestarter.go:41]: nevermind this comment. i shouldn't be reviewing PRs this late on a friday :)

**Comment by joelddiaz** [comment]
lgtm

**Pull request merged**